src/sentry/nodestore/base.py

```diff
@@ -58,39 +65,120 @@ def delete_multi(self, id_list):
         for id in id_list:
             self.delete(id)
 
-    def get(self, id):
+    def _decode(self, value, subkey):
+        if value is None:
+            return None
+
+        lines_iter = iter(value.splitlines())
+        try:
+            if subkey is not None:
+                next(lines_iter)
+
+                for line in lines_iter:
+                    if line.strip() == subkey.encode("utf8"):
+                        break
+
+            return json_loads(next(lines_iter))
+        except StopIteration:
+            return None
+
+    def _get_bytes(self, id):
         """
-        >>> data = nodestore.get('key1')
-        >>> print data
+        >>> nodestore._get_bytes('key1')
+        b'{"message": "hello world"}'
         """
         raise NotImplementedError
 
-    def get_multi(self, id_list):
+    def get(self, id, subkey=None):
         """
-        >>> data_map = nodestore.get_multi(['key1', 'key2')
-        >>> print 'key1', data_map['key1']
-        >>> print 'key2', data_map['key2']
+        >>> nodestore.get('key1')
+        {"message": "hello world"}
         """
-        return dict((id, self.get(id)) for id in id_list)
+        if subkey is None:
```

Comment:
Two options:
- This condition should go if you want to use the cache for every subkey.
- Otherwise, you should only _write_ to the cache if the subkey is `None`.

cc @beezz. Provided that our subkeys are used infrequently and are usually queried from cold caches, should we even cache them?

Reply:
will revert changes to cache layout in any case such that we don't have to invalidate caches

Reply:
In the interest of getting this done fast I will just not have caching for reprocessing payloads. If michal or somebody else disagrees we can revisit this decision in a followup